server/channels/store/sqlstore/user_store.go

```diff
@@ -2265,3 +2266,104 @@ func (us SqlUserStore) RefreshPostStatsForUsers() error {
 
 	return nil
 }
+
+func (us SqlUserStore) GetUserReport(
+	sortColumn string,
+	sortDesc bool,
+	pageSize int,
+	lastSortColumnValue string,
+	lastUserId string,
+	startAt int64,
+	endAt int64,
+	roleFilter string,
+	teamFilter string,
+	hasNoTeam bool,
+	hideActive bool,
+	hideInactive bool,
+) ([]*model.UserReportQuery, error) {
+	isPostgres := us.DriverName() == model.DatabaseDriverPostgres
+	selectColumns := []string{"u.Id", "u.LastLogin", "s.LastActivityAt AS LastStatusAt"}
+	for _, column := range model.UserReportSortColumns {
+		selectColumns = append(selectColumns, "u."+column)
+	}
+	if isPostgres {
+		selectColumns = append(selectColumns,
+			"MAX(ps.LastPostDate) AS LastPostDate",
+			"COUNT(ps.Day) AS DaysActive",
+			"SUM(ps.NumPosts) AS TotalPosts",
+		)
+	} else {
+		selectColumns = append(selectColumns,
+			"MAX(p.CreateAt) AS LastPostDate",
+			"COUNT(DATE(FROM_UNIXTIME(p.CreateAt / 1000))) AS DaysActive",
+			"COUNT(p.Id) AS TotalPosts",
+		)
+	}
+
+	sortColumnValue := sortColumn
+	if sortDesc {
+		sortColumnValue += " DESC"
+	}
+
+	query := us.getQueryBuilder().
+		Select(selectColumns...).
+		From("Users u").
+		LeftJoin("Status s ON s.UserId = u.Id").
+		Where(sq.Or{
+			sq.Gt{sortColumn: lastSortColumnValue},
+			sq.And{
+				sq.Eq{sortColumn: lastSortColumnValue},
+				sq.Gt{"u.Id": lastUserId},
+			},
+		}).
+		Where(sq.Expr("u.Id NOT IN (SELECT UserId FROM Bots)")).
+		GroupBy("u.Id", "s.UserId").
+		OrderBy(sortColumnValue, "u.Id")
+
+	if pageSize > 0 {
+		query = query.Limit(uint64(pageSize))
+	}
+
+	if isPostgres {
+		joinQuery := "PostStats ps ON ps.UserId = u.Id"
+		if startAt > 0 {
+			startDate := time.UnixMilli(startAt)
+			joinQuery += fmt.Sprintf(" AND ps.Day >= '%s'", startDate.Format("2006-01-02"))
+		}
+		if endAt > 0 {
+			endDate := time.UnixMilli(endAt)
+			joinQuery += fmt.Sprintf(" AND ps.Day < '%s'", endDate.Format("2006-01-02"))
+		}
+		query = query.LeftJoin(joinQuery)
+	} else {
+		joinQuery := "Posts p on p.UserId = u.Id"
+		if startAt > 0 {
+			joinQuery += fmt.Sprintf(" AND p.CreateAt >= %d", startAt)
+		}
+		if endAt > 0 {
+			joinQuery += fmt.Sprintf(" AND p.CreateAt < %d", endAt)
+		}
+		query = query.LeftJoin(joinQuery)
+	}
+
+	query = applyRoleFilter(query, roleFilter, isPostgres)
+	if hasNoTeam {
+		query = query.Where(sq.Expr("u.Id NOT IN (SELECT UserId FROM TeamMembers WHERE DeleteAt = 0)"))
+	} else if teamFilter != "" {
+		query = query.Join("TeamMembers tm ON ( tm.UserId = u.Id AND tm.DeleteAt = 0 AND tm.TeamId = ? )", teamFilter)
+	}
+	if hideActive {
+		query = query.Where(sq.Gt{"u.DeleteAt": 0})
+	}
+	if hideInactive {
+		query = query.Where(sq.Eq{"u.DeleteAt": 0})
+	}
```

Comment:
I see we are not validating anywhere that setting both of these to true is invalid. Let's add a validation in the api layer for this.

Reply:
It's still an invalid case right? I think it might be helpful to the user to get a friendly error message rather than 0 rows and then have to figure out where did they go wrong. LMK what you think.